src/xdp-utils.c

```diff
@@ -572,6 +573,75 @@ cleanup_temp_file (void *p)
   g_free (*pp);
 }
 
+#define VALIDATOR_INPUT_FD 3
+
+static GKeyFile *
+spawn_validator (const char* const* args,
+                 int                fd)
+{
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GBytes) bytes_output = NULL;
+  g_autoptr(GKeyFile) key_file = NULL;
+  g_autoptr(GSubprocessLauncher) launcher = NULL;
+  g_autoptr(GMappedFile) mapped_output = NULL;
+  g_autoptr(GSubprocess) process = NULL;
+  xdp_autofd int input_fd = -1;
+  int output_fd = -1;
+  xdp_autofd int stdout_fd = -1;
+
+  stdout_fd = memfd_create("spawn-validator-stdout", 0);
```

Comment:
I don't see the point of using a memfd for stdout. It's a few bytes of text.

Reply:
This let me down a rabbit hole improving the `xdp_spawn ()` and `xdp_spawnv ()`. But i think it's much better now :)